### DISCUSSION TITLE
Loosing all contributions by changing primary email address of github account

### DISCUSSION THREAD

**Discussion by Ammoniya** [community/community#21616] [New to GitHub]
<p>I have added a new email address to GitHub account and deleted my old primary email address from account. Now   organizations don’t show me as a collaborator of repos. But commits are there. And all contributions are gone from my contribution time line of my profile as well. Is there anyway to recover this ?</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>Any commits that used your old primary email address will no longer be associated with your account if you no longer have that email address associated with your account.</p>
<p>To revert, you would have to add that email address back and verify it. It doesn’t have to be your primary email address.</p>
<p>To prevent this becoming a problem in the future, especially if you are using an email address that is attached to a school or workplace, you can choose to select our email privacy feature and use a <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-on-github">GitHub provided address for all of your commits</a>. This isn’t a functional email address, but a way of associating your commits directly with your GitHub account instead of with a specific email address.</p>

**Reply by mortezanourani** [reply]
I have this problem. I added the removed email again but my contribution time line still cleared.
What should I do?

**Reply by ToniBirat7** [reply]
@mortezanourani did you find any fix?? It's been like 12 hrs still my contributions are not seen

**Reply by 5534510294alesan-alt** [reply]
> 👋 Welcome!
> 
> Any commits that used your old primary email address will no longer be associated with your account if you no longer have that email address associated with your account.
> 
> To revert, you would have to add that email address back and verify it. It doesn’t have to be your primary email address.
> 
> To prevent this becoming a problem in the future, especially if you are using an email address that is attached to a school or workplace, you can choose to select our email privacy feature and use a [GitHub provided address for all of your commits](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-on-github). This isn’t a functional email address, but a way of associating your commits directly with your GitHub account instead of with a specific email address.

**Reply by N3xthar** [reply]
dude give me solution how you all fix it and restore you  old contribution dude i want a help dude

**Comment by Ammoniya** [comment]
<p>What if that removed primary email was old and unaccessible , due to that I’m not able to verify the email address.</p>

**Reply by faizanmunir017** [reply]
Same issue. I added that email in my personal account but obviously its unverified due to the lack of access . I actually wanted to move my contributions at work to show up in my personal account but now I am left with neither haha. Please let me know if you have found anything

**Comment by canuckjacq** [comment]
<p>Try adding it without verifying - it might still work.</p>

**Comment by jfullstackdev** [comment]
<p>yup, simply add without verifying it, I did that, it’s working</p>

**Reply by jairus-m** [reply]
How long did it take to work?

**Reply by v1nshul** [reply]
> How long did it take to work?
is it working for you now ?

**Reply by jairus-m** [reply]
@v1nshul it worked for public repos within 24 hours (does not show I am contributor yet) but am not seeing the majority of my commits made to private repos for a private organization. i raised a ticket and GitHub support is working on it!

**Comment by Ammoniya** [comment]
<p><a class="mention" href="/canuckjacq">@canuckjacq</a> Thanks , <a class="mention" href="/jdevstatic">@jdevstatic</a> yes it is, thanks</p>

**Comment by jfullstackdev** [comment]
<p><a class="mention" href="/canuckjacq">@canuckjacq</a>, is this not a security hole? What if, for example, I know a certain programmer and the email account he is using, if I want to get his commits, I can do so by simply adding the email without verifying it, right? But, of course, I will not do that, I’m just exposing the security issue here.</p>
<p>Unless, GitHub marks emails that should be used once, for example, if my old email is added to my account and I cannot access that anymore but I added it without verifying it, then another user used that also after the authentic user added it, the second user will not be allowed because the email is already associated with the valid GitHub user, even though he did not verify it. Hopefully, this will be added if there is none.</p>

**Comment by canuckjacq** [comment]
<p>I can’t see where it would be a security risk. If someone is super hung up on clocking up meaningless commits there are definitely simpler ways to go about that using your own email address.</p>
<p>You can’t add an email address to your account if it is already on another account - verified or not. Additionally, if someone adds your email address to their GitHub account, you can ask support to remove it.</p>

**Comment by jfullstackdev** [comment]
<aside class="quote group-github-employees" data-username="canuckjacq" data-post="9" data-topic="179135">
<div class="title">
<div class="quote-controls"></div>
 canuckjacq:</div>
<blockquote>
<p>You can’t add an email address to your account if it is already on another account - verified or not.</p>
</blockquote>
</aside>
<p>this is what I’m referring to, this is great so that it’s sure, this was not mentioned in the documentation, thanks for the reply,</p>

**Comment by EtoDemerzel0427** [comment]
@canuckjacq Does it take time to get the commits back? I removed my primary email the other day, after I noticed I lost my commits I added it back, that was one day ago but till now I still don't see my lost commits back.

**Comment by vatsa287** [comment]
@EtoDemerzel0427 faced the same issue here while changing primary address. I deleted the email address, the saw all contributions count made from that email was lost. Added the email back, verified it and made it as a primary email account. Able to see all contribution count back. Key here is to make it primary again. Later when you see the contribution count change the primary email as per your needs, just do not delete that account from github. Its ok to keep it un-verified for the commits to be seen, but do not delete it.

**Reply by JOS-RE** [reply]
Hi! I tried doing the same and its been around a month already, but do not see any of my past contributions coming back. Hi! I tried doing the same and its been around a month already, but do not see any of my past contributions coming back.

**Comment by EtoDemerzel0427** [comment]
@vatsa287 Yes I did see it back, though it took three days in my case, not sure why it was so long.

**Comment by KimKyungup** [comment]
T.T

**Comment by Satarupa22-SD** [comment]
I have signed up for GitHub using my edu email what will happen once I no longer have access to it. Will I lose my account or contributions completely? Is there any way around to prevent it. Initially I had signed up using edu email to avail the GitHub developer benefits. What should I do now. I have added a backup account.

**Reply by aceof10** [reply]
If you initially used your edu email for commits, and you remove it, you will lose contributions associated with that email.
You can choose to use the noreply email provided by Github for your commits. This ensures that commits are associated with that particular Github account and not a particular email.
[[GitHub provided address for all of your commits](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-on-github)]

Afterwards, change your primary email address to any other email of your choice.

**Comment by Arzzam** [comment]
I normally use github private emails to commit in my machine, when I brought a new machine, I unfortunately set my personal email id and I made some contribution in some repos, and then I realise I use private email for github. I contributed to the repos but My contributions are not available or not shown in the contributors tab in that repos, Is there any way other way to retrive that.
Thanks.

**Comment by notstarboard** [comment]
I am in the midst of replacing all of my email addresses with aliases for privacy reasons. Because GitHub insists on tying commits to email addresses instead of to accounts, it seems I will either need to butcher all of my old metadata to change those commits to my alias or those commits will no longer be associated with my account. I want all of my old commits to be associated with my account without my actual email address being associated with my account. This can't be such an uncommon use case.

Someone please tell me if I'm missing something, but this seems like an absolute mess of a system.

**Comment by insooneelife** [comment]
I did all things to recover the contributions, but the private commits still does not show up.

**Comment by gklrajan** [comment]
@insooneelife @EtoDemerzel0427 @mortezanourani : to the ones who added back their old email but still can't see the commits - i noted that making at least 1 commit in the repos where your contributions are missing will restore all your contributions in that repo. you will have to do for every repo that you want to recover on your contributions! good luck!

**Reply by CharlesPerrotMinot** [reply]
Tried that, didn't work unfortunately

**Reply by GROOOOAAAARK** [reply]
Your solution seems to work for me, thanks a lot @gklrajan ! Been wrapping my head around this issue like crazy lately

**Reply by CharlesPerrotMinot** [reply]
There's probably a threshold, looks like it's not done for large repositories

**Comment by shulaoda** [comment]
No!!! 😭😭😭

**Reply by shulaoda** [reply]
All contributions are back — false alarm. 😀

**Reply by kibinzxc** [reply]
how many hours/days did it took before it goes back? did you verify it?

**Comment by viniciushgiovanini** [comment]
I removed my main email, and later I added it back and verified it. The repository history reappeared, but my heatmap didn’t come back. Now idk how I can get them back.

**Reply by lst4rk** [reply]
did it ever come back?

**Reply by viniciushgiovanini** [reply]
I was only able to restore my repository history, but the contribution heatmap on my profile didn’t come back. I gave up on restoring that part, but I saw in the comment below that someone managed to do it by opening a ticket with GitHub Support.

**Comment by tonihoo** [comment]
I opened a ticket in Github support and they restored my lost contributions within a few hours. I had removed an old email where I no longer had access to, and re-adding that (as unverified) did not help. But now all contributions appear again, nice!

**Reply by viniciushgiovanini** [reply]
Were you able to restore your profile heatmap ?

**Reply by tonihoo** [reply]
Yes!

**Reply by viniciushgiovanini** [reply]
Thank you for the tip. I managed to restore my commit history and my commit heatmap

**Comment by viniciushgiovanini** [comment]
Dear friends, **I managed to return my commit history  and my commit heatmap**. I returned the email in witch I made the commits, this returned the commit history links to my account in each repositorory. When I did this, the activity map in my profile didn't return. I opened a ticket reporting this and they fixed it. #tip

**Reply by abdullahalhakimi** [reply]
how did you open the ticket?

**Reply by viniciushgiovanini** [reply]
> how did you open the ticket?

Search github support on google --> my ticket --> new ticket

**Comment by Chara-Freedom** [comment]
I can confirm that writing to the support is helping very much

**Comment by EN-AenaHabib** [comment]
Nothing is lost.

GitHub only shows contributions if the commit email matches a verified email on your account.

Fix: Go to Settings → Emails, add and verify your old email again (even temporarily).

After that, your commits will reconnect, and your contribution graph and org visibility will return.
